server.go

```diff
@@ -163,6 +163,7 @@ func OptServerGCNotifier(gcn GCNotifier) ServerOption {
 func OptServerRemoteClient(c *http.Client) ServerOption {
 	return func(s *Server) error {
 		s.executor = NewExecutor(c)
+		s.remoteClient = c
```

Comment:
I think we should have defaults for all Server fields, including `remoteClient` in case the user doesn't call option functions.